copy/copy.go

```diff
@@ -394,17 +396,36 @@ func shortDigest(d digest.Digest) string {
 	return d.Encoded()[:12]
 }
 
-// createProgressBar creates a pb.ProgressBar.
-func createProgressBar(srcInfo types.BlobInfo, kind string, writer io.Writer) *pb.ProgressBar {
+// createProgressBar creates a pb.ProgressBar.  Note that if the copier's
+// reportWriter is a file but a tty, the progress bar's output will be
+// discarded and a single line will be printed instead.
+func (c *copier) createProgressBar(srcInfo types.BlobInfo, kind string) *pb.ProgressBar {
 	bar := pb.New(int(srcInfo.Size)).SetUnits(pb.U_BYTES)
 	bar.SetMaxWidth(80)
 	bar.ShowTimeLeft = false
 	bar.ShowPercent = false
-	bar.Prefix(fmt.Sprintf("Copying %s %s:", kind, shortDigest(srcInfo.Digest)))
-	bar.Output = writer
+	prefix := fmt.Sprintf("Copying %s %s:", kind, shortDigest(srcInfo.Digest))
+	bar.Prefix(prefix)
```

Comment:
(The new `prefix` variable seems to have no other users, were there supposed to be any?)

Reply:
Artefact of tiredness, sorry for that.